raft/src/test/java/org/apache/kafka/raft/KafkaRaftClientTest.java

```diff
@@ -97,14 +100,22 @@ private InetSocketAddress mockAddress(int id) {
     }
 
     private KafkaRaftClient buildClient(Set<Integer> voters) throws IOException {
+        return buildClient(voters, new MockStateMachine());
```

Comment:
could we just pass in the parameter `stateMachine` here? We already have the cleanup logic after each run, so it should be safe.